File: samples/js/src/agents/policycheck/test_client.mjs
#!/usr/bin/env node
/**
 * PolicyCheck A2A Test Client
 *
 * Demonstrates interacting with a live hosted A2A service agent.
 * No dependencies required — uses native fetch.
 *
 * Usage: node test_client.mjs
 */

const A2A_ENDPOINT = "https://legaleasy.tools/api/a2a";
const AGENT_CARD_URL = "https://legaleasy.tools/.well-known/agent.json";

let rpcId = 0;

async function sendA2ARequest(message) {
  const body = {
    jsonrpc: "2.0",
    id: ++rpcId,
    method: "message/send",
    params: { message },
  };

  const res = await fetch(A2A_ENDPOINT, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  });

  if (!res.ok) throw new Error(`HTTP ${res.status}: ${await res.text()}`);
  return res.json();
}

function printResult(label, response) {
  console.log(`\n${"=".repeat(60)}`);
  console.log(label);
  console.log("=".repeat(60));

  if (response.error) {
    console.log("ERROR:", response.error.message);
    return;
  }

  const task = response.result;
  console.log(`Task ID:  ${task.id}`);
  console.log(`Status:   ${task.status.state}`);

  // Print human-readable summary from status message
  const statusText = task.status?.message?.parts?.find((p) => p.kind === "text");
  if (statusText) {
    console.log(`\nSummary:\n${statusText.text}`);
  }

  // Print structured data from artifacts
  const dataArtifact = task.artifacts?.[0]?.parts?.find((p) => p.kind === "data");
  if (dataArtifact) {
    const d = dataArtifact.data;
    console.log(`\nStructured Data:`);
    console.log(`  Risk Level:            ${d.riskLevel}`);
    console.log(`  Buyer Protection Score: ${d.buyerProtectionScore}/100`);
    console.log(`  Recommendation:        ${d.recommendation}`);
    if (d.policiesFound) {
      console.log(`  Policies Found:        ${d.policiesFound.join(", ")}`);
    }
  }
}

async function main() {
  console.log("PolicyCheck A2A Test Client");
  console.log("Endpoint:", A2A_ENDPOINT);

  // ── Test 1: Discover agent card ──────────────────────────────────────────
  console.log(`\n${"=".repeat(60)}`);
  console.log("Test 1: Agent Card Discovery");
  console.log("=".repeat(60));

  const cardRes = await fetch(AGENT_CARD_URL);
  if (!cardRes.ok) {
    console.log(`ERROR: Could not fetch agent card (HTTP ${cardRes.status})`);
  } else {
    const card = await cardRes.json();
    console.log(`Name:         ${card.name}`);
    console.log(`Description:  ${card.description}`);
    console.log(`URL:          ${card.url}`);
    console.log(`Skills:       ${card.skills?.map((s) => s.id).join(", ")}`);
    console.log(`Version:      ${card.version}`);
  }

  // ── Test 2: Natural language analysis ────────────────────────────────────
  const nlResponse = await sendA2ARequest({
    role: "user",
    parts: [
      { kind: "text", text: "Quick check on https://www.amazon.com" },
    ],
  });
  printResult("Test 2: Natural Language Request (Amazon quick check)", nlResponse);

  // ── Test 3: Structured quick risk check ──────────────────────────────────
  const structuredResponse = await sendA2ARequest({
    role: "user",
    parts: [
      {
        kind: "data",
        data: { seller_url: "https://www.amazon.com", skill: "quick-risk-check" },
        mimeType: "application/json",
      },
    ],
  });
  printResult("Test 3: Structured Data Request (quick-risk-check)", structuredResponse);

  // ── Test 4: Raw policy text with known risks ─────────────────────────────
  const riskyText =
    "All sales are final. No refunds. By using this service you agree to " +
    "binding arbitration and waive your right to class action lawsuits. " +
    "Our liability shall not exceed $50. Your account may be terminated " +
    "at any time without notice. This subscription will auto-renew annually.";

  const textResponse = await sendA2ARequest({
    role: "user",
    parts: [{ kind: "text", text: riskyText }],
  });
  printResult("Test 4: Raw Policy Text (risky terms)", textResponse);

  console.log(`\n${"=".repeat(60)}`);
  console.log("All tests complete.");
  console.log("=".repeat(60));
}

main().catch((err) => {
  console.error("Fatal error:", err);
  process.exit(1);
});

File: samples/python/agents/dnai_asha_medical/test_client.py
"""
DNAi Asha A2A Test Client
==========================
Demonstrates connecting to the hosted Asha medical agent via A2A protocol.

Usage:
    ASHA_API_KEY=your_key uv run test_client.py
"""

import json
import os
import sys

import httpx

BASE_URL = "https://api.askasha.org"
API_KEY = os.getenv("ASHA_API_KEY", "")


def discover():
    """Fetch and display the agent card."""
    with httpx.Client(timeout=10) as client:
        r = client.get(f"{BASE_URL}/.well-known/agent-card.json", params={"agent_id": "asha"})
        r.raise_for_status()
        card = r.json()

    print(f"Agent: {card['name']} v{card['version']}")
    print(f"Provider: {card['provider']['organization']}")
    print(f"Protocol: {card['supportedInterfaces'][0]['protocolVersion']}")
    print("Skills:")
    for skill in card["skills"]:
        print(f"  [{skill['id']}] {skill['name']}")
    if "epistemicCapabilities" in card:
        ec = card["epistemicCapabilities"]
        corpus = ec.get("knowledgeCorpus", {})
        print(f"Knowledge: {corpus.get('totalVectors', '?')} vectors, {len(corpus.get('collections', []))} collections")
    return card


def query(text: str) -> dict:
    """Send a medical question via A2A."""
    with httpx.Client(timeout=120) as client:
        r = client.post(
            f"{BASE_URL}/a2a/v1/message:send",
            headers={"Authorization": f"Bearer {API_KEY}", "Content-Type": "application/json"},
            json={
                "message": {"role": "user", "parts": [{"text": text}]},
                "metadata": {"agent_id": "asha"},
            },
        )
        r.raise_for_status()
        return r.json()["task"]


def main():
    if not API_KEY:
        print("Set ASHA_API_KEY. Get a free key:")
        print("  curl -X POST https://api.askasha.org/api/a2a/signup \\")
        print('    -H "Content-Type: application/json" \\')
        print('    -d \'{"email":"you@example.com","name":"Test","tier":"free"}\'')
        sys.exit(1)

    print("=" * 50)
    print("DNAi Asha — A2A Medical Agent")
    print("=" * 50)

    discover()

    print("\n--- Query ---")
    task = query("What are the interactions between warfarin and amiodarone?")
    print(f"State: {task['status']['state']}")

    for artifact in task.get("artifacts", []):
        if artifact.get("name") == "response" and artifact.get("parts"):
            print(f"Answer: {artifact['parts'][0]['text'][:300]}...")
        elif artifact.get("name") == "provenance" and artifact.get("parts"):
            prov = artifact["parts"][0].get("data", {})
            print(f"Sources: {prov.get('sources', [])}")
            print(f"Evidence: {prov.get('evidence_count', 0)} items")


if __name__ == "__main__":
    main()
